db/memtable.cc

```diff
@@ -571,13 +575,18 @@ static bool SaveValue(void* arg, const char* entry) {
         }
         Slice v = GetLengthPrefixedSlice(key_ptr + key_length);
         *(s->status) = Status::OK();
-        if (*(s->merge_in_progress)) {
-          *(s->status) = MergeHelper::TimedFullMerge(
-              merge_operator, s->key->user_key(), &v,
-              merge_context->GetOperands(), s->value, s->logger, s->statistics,
-              s->env_);
-        } else if (s->value != nullptr) {
-          s->value->assign(v.data(), v.size());
+        if (LIKELY(s->pSlice != nullptr)) {
+          if (*(s->merge_in_progress)) {
+            *(s->status) = MergeHelper::TimedFullMerge(
+                merge_operator, s->key->user_key(), &v,
+                merge_context->GetOperands(), s->pSlice->GetSelf(), s->logger,
+                s->statistics, s->env_);
+            s->pSlice->PinSelf();
+          } else {
+            //s->mem->Ref();
+            //s->pSlice->PinSlice(v, UnrefMemTable, s->mem, nullptr);
+            s->pSlice->PinSelf(v);
```

Comment:
Discussed that offline with Islam. The plan is go ahead with the string copy from memtable and revisit the alternatives in future.